device/ethernet.yaml

```diff
@@ -1,58 +1,84 @@
-components:
-  schemas:
-    Device.EthernetBase:
-      x-include: 
-      - '../common/common.yaml#/components/schemas/Named.Object'
-      description: >-
-        Base Ethernet interface.
-      type: object
-      required: [mac]
-      properties:
-        mac:
-          description: >-
-            Media Access Control address.
-          type: string
-          format: mac
-        mtu:
-          description: >-
-            Maximum Transmission Unit.
-          type: integer
-          minimum: 0
-          maximum: 65535
-          default: 1500
-        vlans:
-          description: >-
-            List of VLANs
-          type: array
-          items:
-            $ref: './vlan.yaml#/components/schemas/Device.Vlan'
-
-    Device.Ethernet:
-      x-include: 
-      - '#/components/schemas/Device.EthernetBase'
-      description: >-
-        An Ethernet interface with IPv4 and IPv6 addresses.
-      type: object
-      required: [port_name]
-      properties:
-        port_name: 
-          description: |-
-            The unique name of a Port or a LAG that will emulate this interface.
-          type: string
-          x-constraint:
-          - '/components/schemas/Port/properties/name'
-          - '/components/schemas/Lag/properties/name' 
-        ipv4_addresses:
-          description: |-
-            List of IPv4 addresses and their gateways. 
-          type: array
-          items:
-            $ref: './ipv4.yaml#/components/schemas/Device.Ipv4' 
-        ipv6_addresses:
-          description: |-
-            List of global IPv6 addresses and their gateways.
-            The Link Local IPv6 address will be automatically generated.
-          type: array
-          items:
-            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6'
-      
\ No newline at end of file
+components:
+  schemas:
+    Device.EthernetBase:
+      x-include: 
+      - '../common/common.yaml#/components/schemas/Named.Object'
+      description: >-
+        Base Ethernet interface.
+      type: object
+      required: [mac]
+      properties:
+        mac:
+          description: >-
+            Media Access Control address.
+          type: string
+          format: mac
+        mtu:
+          description: >-
+            Maximum Transmission Unit.
+          type: integer
+          minimum: 0
+          maximum: 65535
+          default: 1500
+        vlans:
+          description: >-
+            List of VLANs
+          type: array
+          items:
+            $ref: './vlan.yaml#/components/schemas/Device.Vlan'
+
+    Device.Ethernet:
+      x-include: 
+      - '#/components/schemas/Device.EthernetBase'
+      description: >-
+        An Ethernet interface with IPv4 and IPv6 addresses.
+      type: object
+      properties:
+        connection: 
+          description: |-
+            Device connection to physical, LAG or another device.
+          $ref : '#/components/schemas/Device.Connection'
+        port_name: 

```

Comment:
I am assuming that this will be marked as deprecated and the description will be updated?

Reply:
updated